test/prism/errors_test.rb

```diff
@@ -64,6 +64,24 @@ def test_invalid_message_name
       assert_equal :"", Prism.parse_statement("+.@foo,+=foo").write_name
     end
 
+    def test_regexp_encoding_option_mismatch_error
+      # UTF-8 char with ASCII-8BIT modifier
+      result = Prism.parse('/Ȃ/n')
+      assert_includes result.errors.map(&:type), :regexp_encoding_option_mismatch
+
+      # UTF-8 char with EUC-JP modifier
+      result = Prism.parse('/Ȃ/e')
+      assert_includes result.errors.map(&:type), :regexp_encoding_option_mismatch
+
+      # UTF-8 char with Windows-31J modifier
+      result = Prism.parse('/Ȃ/s')
+      assert_includes result.errors.map(&:type), :regexp_encoding_option_mismatch
+
+      # UTF-8 char with UTF-8 modifier
+      result = Prism.parse('/Ȃ/u')
+      assert_empty result.errors
+    end
+
```

Comment:
should this test be added to `prism/encoding/regular_expression_encoding_test.rb` instead?

Reply:
It's fine here.